src/app/qgshandlebadlayers.cpp

```diff
@@ -450,16 +444,11 @@ void QgsHandleBadLayers::apply()
       else
       {
         item->setForeground( QBrush( Qt::red ) );
-        if ( mFileBase[ name ].size() == 1 )
-          mFileBase[ name ][0] = basepath ;
-        else if ( mFileBase[ name ].size() > 1 )
-          mFileBase[ name ].append( basepath );
       }
     }
   }
 
-  // Final cleanup: remove any remaining bad layer
-  // (there should not be any at this point)
+  // Final cleanup: remove any bad layer (it should not be any btw)
```

Comment:
```suggestion
  // Final cleanup: remove any bad layer (it should not be any by now)
```

Reply:
rephrased it, let me know if it's better.